src/Moryx.Notifications/Adapter/NotificationAdapter.cs

```diff
@@ -75,11 +75,9 @@ public void Publish(INotificationSender sender, INotification notification, obje
             }
 
             _listLock.EnterWriteLock();
-            
-            var managed = (IManagedNotification)notification;
-            managed.Identifier = Guid.NewGuid();
-            managed.Created = DateTime.Now;
-            managed.Sender = sender.Identifier;
+
+            notification.Created = DateTime.Now;
```

Comment:
It looks like that the identifier is missing like in managed.Idetifier = Guid.NewGuid();